src/NuGet.Clients/NuGet.PackageManagement.UI/Options/AddMappingDialog.xaml.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Windows;
+using System.Windows.Input;
+using Microsoft.Build.Utilities;
+using Microsoft.ServiceHub.Framework;
+using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.VisualStudio.Common;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
+using NuGet.Configuration;
+using System.Collections.Generic;
+using Task = System.Threading.Tasks.Task;
+using Resx = NuGet.PackageManagement.UI.Resources;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+
+using Microsoft.VisualStudio.Shell;
+
+
+namespace NuGet.Options
+{
+    public partial class AddMappingDialog : Window
+    {
+        public ICommand HideButtonCommand { get; set; }
+
+        public ICommand AddButtonCommand { get; set; }
+
+        public ItemsChangeObservableCollection<PackageSourceItem> SourcesCollection { get; private set; } //change to package s
+
+        private IReadOnlyList<PackageSourceContextInfo> _originalPackageSources;
+
+#pragma warning disable ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+        private INuGetSourcesService _nugetSourcesService;
+#pragma warning restore ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+
+        private PackageSourceMappingOptionsControl _parent;
+
+        public AddMappingDialog(PackageSourceMappingOptionsControl parent)
+        {
+            _parent = parent;
+            HideButtonCommand = new ButtonCommand(ExecuteHideButtonCommand, CanExecuteHideButtonCommand);
+            AddButtonCommand = new ButtonCommand(ExecuteAddButtonCommand, CanExecuteAddButtonCommand);
+            SourcesCollection = new ItemsChangeObservableCollection<PackageSourceItem>();
+            DataContext = this;
+            InitializeComponent();
+            CancellationToken cancellationToken = new CancellationToken(false);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => await InitializeOnActivatedAsync(cancellationToken));
+        }
+
+        internal async Task InitializeOnActivatedAsync(CancellationToken cancellationToken)
+        {
+            IServiceBrokerProvider serviceBrokerProvider = await ServiceLocator.GetComponentModelServiceAsync<IServiceBrokerProvider>();
+            IServiceBroker serviceBroker = await serviceBrokerProvider.GetAsync();
+#pragma warning disable ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+            _nugetSourcesService = await serviceBroker.GetProxyAsync<INuGetSourcesService>(
+                    NuGetServices.SourceProviderService,
+                    cancellationToken: cancellationToken);
+#pragma warning restore ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+
+            //show package sources on open
```

Comment:
Right now I am showing all sources (enabled and disabled). Should I not show disabled sources? Or maybe show disabled sources but add something saying they are disabled?

Reply:
Maybe an option could be to show all the sources and have the disabled ones be grayed out + a disabled tag to indicate that they are disabled.

Reply:
https://github.com/NuGet/Client.Engineering/issues/1832